From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Eric Lundby <Eric.Lundby@gmail.com>
Date: Wed, 21 Oct 2026 09:50:17 -0600
Subject: [PATCH] Handle Pillow Image.getdata deprecation

Use Image.get_flattened_data() when available while retaining support for
older Pillow versions.

Backport only the PIL plugin change from:
https://github.com/scikit-image/scikit-image/pull/8010
https://github.com/scikit-image/scikit-image/commit/9d8daba419249f9d3327fe85177d79b18adb6032

---
 src/skimage/io/_plugins/pil_plugin.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/skimage/io/_plugins/pil_plugin.py b/src/skimage/io/_plugins/pil_plugin.py
index 7a75c5f..098d22a 100644
--- a/src/skimage/io/_plugins/pil_plugin.py
+++ b/src/skimage/io/_plugins/pil_plugin.py
@@ -47,6 +47,10 @@ def pil_to_ndarray(image, dtype=None, img_num=None):
     Refer to ``imread``.
 
     """
+    # PIL 12.1.0 renames getdata
+    if hasattr(image, "get_flattened_data"):
+        image.getdata = image.get_flattened_data
+
     try:
         # this will raise an IOError if the file is not readable
         image.getdata()[0]
-- 
2.50.1
